src/backend/distributed/sql/udfs/citus_stat_activity/latest.sql

```diff
@@ -0,0 +1,23 @@
+DROP VIEW IF EXISTS pg_catalog.citus_stat_activity;
+DROP TYPE IF EXISTS pg_catalog.global_pid_and_pg_stat_activity;
+
+CREATE TYPE citus.global_pid_and_pg_stat_activity AS (global_pid bigint, pg_stat_activity pg_stat_activity);
+
+CREATE VIEW citus.citus_stat_activity AS
+SELECT gpid_and_psa_from_all_nodes.global_pid, nodeid, (gpid_and_psa_from_all_nodes.pg_stat_activity).*
+FROM (
+    SELECT * FROM run_command_on_all_nodes($$
+        SELECT coalesce(to_json(array_agg(gpid_and_psa.*)), '[{}]'::JSON)
+        FROM (
+            SELECT global_pid, (pg_stat_activity.*)::pg_stat_activity FROM
+            pg_stat_activity LEFT JOIN get_all_active_transactions() ON process_id = pid
+        ) AS gpid_and_psa;
+    $$)
+    WHERE success = 't'
+) AS run_command_on_all_nodes
+LEFT JOIN LATERAL json_populate_recordset(NULL::global_pid_and_pg_stat_activity, run_command_on_all_nodes.result::JSON) as gpid_and_psa_from_all_nodes ON TRUE;
```

Comment:
Sorry, I misguided you in the chat. I think this join should not be a LEFT JOIN, that seems redundant